File: src/main/java/Day2/Day2.js
var passwords ;
var fs = require("fs");
function loadFile() {
  //  var text = fs.readFileSync("Day1_1.txt").toString('utf-8');
  var text = fs.readFileSync("resources/Day2.txt").toString('utf-8');
    passwords = text.split("\n");
}

loadFile();
var count =0;
var number =0;

function findBookings(passwords){

 for( const password of passwords){
   var arr = password.split(/[-: ]+/)
   var boundar =0;
jdfa;d{
}

   var min = arr[0];
   var max = arr[1];
   var character = arr[2]

   var passcode = arr[3]



   var answer = arr[3].length-(arr[3].replaceAll(arr[2], "")).length;
   if(passcode.charAt(max-1)===character){
     boundar++;

   }
   if(passcode.charAt(min-1)===character){
     boundar++
   }
   if(boundar===1){
     count++
   }
   if(answer<=max && answer>=min){
     number++;
   }


 }

 console.log(count +"answer to 2nd part")
 console.log("first part "+number)
}
findBookings(passwords);
fs.writeFile('answerday2.txt', count +" answer to 2nd part", (error) => {

     // In case of a error throw err exception.
     if (error) throw err;
 })

File: src/main/java/Day2/day2.java
package Day2;

import Utils.ReUtils1;

import java.io.IOException;
import java.util.Arrays;

public class day2 {

    public static void main(String[] args) throws IOException {

        ReUtils1 utils = new ReUtils1();
        split(utils.readFile("2") );
        onlyone(utils.readFile("2"));
    }


    public static int split(String[] input) {
        int count=0;
        for (String b : input) {
            String[] answer = b.split(",");
            for (String s : answer) {
                String[] pass = s.split(":|-|\\s");
                int number = pass[4].length() -pass[4].replaceAll(pass[2],"").length();
                if(number >= Integer.valueOf(pass[0]) && number <= Integer.valueOf(pass[1])){
                    count++;
                }


            }


        }
        System.out.println(count +" count");
        return count;

    }

    public static int onlyone(String[] input) {
        int count=0;
        for (String b : input) {
            String[] answer = b.split(",");
            for (String s : answer) {
                String[] pass = s.split(":|-|\\s");
                int min = Integer.valueOf(pass[0]), max = Integer.valueOf(pass[1]);
                int number = pass[4].length() -pass[4].replaceAll(pass[2],"").length();
                if((pass[4].charAt(min-1)== pass[2].charAt(0)&pass[4].charAt(max-1)!= pass[2].charAt(0))||(pass[4].charAt(min-1)!= pass[2].charAt(0)&pass[4].charAt(max-1)== pass[2].charAt(0))){

                    count++;
                }
            }



        }
        System.out.println(count +" count");
        return count;

    }
}

